File: classes/parte-5-funciones-y-modularidad/084-funciones-puras-y-efectos-secundarios/implementaciones/c/main.c
#include <stdio.h>

long cuadrado(long n) {
    return n * n;
}

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    printf("puro=%ld\n", cuadrado(n));
    return 0;
}

File: classes/parte-5-funciones-y-modularidad/084-funciones-puras-y-efectos-secundarios/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func cuadrado(n int64) int64 {
	return n * n
}

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.ParseInt(strings.TrimSpace(line), 10, 64)
	fmt.Printf("puro=%d\n", cuadrado(n))
}
